managed/resource_vpc_peering.go

```diff
@@ -41,49 +41,57 @@ func (r resourceVPCPeeringType) GetSchema(_ context.Context) (tfsdk.Schema, diag
 			},
 			"yugabytedb_vpc_id": {
-				Description: "The ID of the VPC where the YugabyteDB cluster is deployed.",
-				Type:        types.StringType,
-				Required:    true,
+				Description:   "The ID of the VPC where the YugabyteDB cluster is deployed.",
+				Type:          types.StringType,
+				Required:      true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 			},
 			"application_vpc_info": {
 				Description: "The details for the VPC where the application is deployed.",
 				Required:    true,
 				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
 					"cloud": {
-						Description: "The cloud provider (AWS, AZURE or GCP) where the application is deployed.",
-						Type:        types.StringType,
-						Required:    true,
+						Description:   "The cloud provider (AWS, AZURE or GCP) where the application is deployed.",
+						Type:          types.StringType,
+						Required:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 					},
 					"project": {
-						Description: "The project ID for GCP.",
-						Type:        types.StringType,
-						Optional:    true,
+						Description:   "The project ID for GCP.",
+						Type:          types.StringType,
+						Optional:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 					},
 					"account_id": {
-						Description: "The account ID for AWS.",
-						Type:        types.StringType,
-						Optional:    true,
+						Description:   "The account ID for AWS.",
+						Type:          types.StringType,
+						Optional:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 					},
 					"region": {
-						Description: "The region where the application is deployed.",
-						Type:        types.StringType,
-						Optional:    true,
+						Description:   "The region where the application is deployed.",
+						Type:          types.StringType,
+						Optional:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 					},
 					"vpc_id": {
-						Description: "The ID of the VPC in which the application is deployed.",
-						Type:        types.StringType,
-						Required:    true,
+						Description:   "The ID of the VPC in which the application is deployed.",
+						Type:          types.StringType,
+						Required:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
 					},
 					"cidr": {
-						Description: "The CIDR of the VPC in which the application is deployed.",
-						Type:        types.StringType,
-						Computed:    true,
-						Optional:    true,
+						Description:   "The CIDR of the VPC in which the application is deployed.",
+						Type:          types.StringType,
+						Computed:      true,
+						Optional:      true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{tfsdk.RequiresReplace()},
```

Comment:
@shinji62 maybe we can throw a warning about the delete? Things may stop working for a while right?

Reply:
When you do a `terraform apply`, then it will tell you the plan and that it is replacing the resource (it will have a `-` for the old one and a `+` for the new one).

Reply:
as @cdavid Terraform will tell you that a replacement is needed.